src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,45 @@
 import React from 'react';
+import { Post } from '../../types/Post';
+import { UserInfo } from '../UserInfo';
+import { CommentList } from '../CommentList';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+import './PostInfo.scss';
+
+interface Props {
+  post: Post,
+}
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    title,
+    user,
+    body,
+    comments,
+  } = post;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">{title}</h3>
+
+        {user && (
+          <p>
+            <UserInfo user={user} />
+          </p>
+        )}
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      <hr />
+
+      {
+        comments.length !== 0
+          ? <CommentList comments={comments} />
+          : <b data-cy="NoCommentsMessage">No comments yet</b>
+      }
```

Comment:
let's keep it concise
```suggestion
      {comments.length > 0
          ? <CommentList comments={comments} />
          : <b data-cy="NoCommentsMessage">No comments yet</b>}
```